pkg/agent/externalnode/external_node_controller.go

```diff
@@ -0,0 +1,751 @@
+				klog.ErrorS(err, "failed to delete uplink port, manual cleanup is required", "portUUID", uplinkUUID, "uplinkName", uplinkName)
+			}
+		}
+	}()
+	uplinkOFPort, ovsErr := c.ovsBridgeClient.GetOFPort(uplinkName, false)
+	if ovsErr != nil {
+		return nil, ovsErr
+	}
+	klog.InfoS("Added uplink port in OVS", "port", uplinkOFPort, "uplinkName", uplinkName)
+
+	// Create host port in OVS.
+	attachInfo := GetOVSAttachInfo(uplinkName, uplinkUUID, eeName, eeNamespace, ips)
+	hostIfUUID, ovsErr := c.ovsBridgeClient.CreateInternalPort(hostIFName, 0, adapterConfig.MAC.String(), attachInfo)
+	if ovsErr != nil {
+		return nil, ovsErr
+	}
+	defer func() {
+		if !success {
+			if ovsErr = c.ovsBridgeClient.DeletePort(hostIfUUID); ovsErr != nil {
+				klog.ErrorS(err, "failed to delete host interface port, manual cleanup is required", "portUUID", hostIfUUID, "hostIFName", hostIFName)
+			}
+		}
+	}()
+	hostOFPort, ovsErr := c.ovsBridgeClient.GetOFPort(hostIFName, false)
+	if ovsErr != nil {
+		return nil, ovsErr
+	}
+	klog.InfoS("Added host port in OVS", "port", hostOFPort, "hostIFName", hostIFName)
+
+	if err = util.MoveIFConfigurations(adapterConfig.IPs, adapterConfig.Routes, adapterConfig.MTU, uplinkName, hostIFName); err != nil {
+		return nil, err
+	}
+	klog.InfoS("Moved configuration from the uplink to host port", "uplink", uplinkName, "hostInterface", hostIFName)
+	if err = c.ofClient.InstallVMUplinkFlows(hostIFName, hostOFPort, uplinkOFPort); err != nil {
+		return nil, err
+	}
+	klog.InfoS("Added uplink and host port in OVS and installed openflow entries", "uplink", uplinkName, "hostInterface", hostIFName)
+	success = true
+	ifIPs := make([]net.IP, 0)
+	for _, ip := range ips {
+		ifIPs = append(ifIPs, net.ParseIP(ip))
+	}
+	hostIFConfig := &interfacestore.InterfaceConfig{
+		Type:          interfacestore.ExternalEntityInterface,
+		InterfaceName: hostIFName,
+		IPs:           ifIPs,
+		OVSPortConfig: &interfacestore.OVSPortConfig{
+			PortUUID: hostIfUUID,
+			OFPort:   hostOFPort,
+		},
+		EntityInterfaceConfig: &interfacestore.EntityInterfaceConfig{
+			EntityName:      eeName,
+			EntityNamespace: eeNamespace,
+			UplinkPort: &interfacestore.OVSPortConfig{
+				PortUUID: uplinkUUID,
+				OFPort:   uplinkOFPort,
+			},
+		},
+	}
+	return hostIFConfig, nil
+}
+
+func GetOVSAttachInfo(uplinkName, uplinkUUID, entityName, entityNamespace string, ips []string) map[string]interface{} {
+	attachInfo := map[string]interface{}{
+		interfacestore.AntreaInterfaceTypeKey: interfacestore.AntreaHost,
+	}
+	if uplinkName != "" {
+		attachInfo[ovsExternalIDUplinkName] = uplinkName
+	}
+	if uplinkUUID != "" {
+		attachInfo[ovsExternalIDUplinkPort] = uplinkUUID
+	}
+	if entityName != "" {
+		attachInfo[ovsExternalIDEntityName] = entityName
+	}
+	if entityNamespace != "" {
+		attachInfo[ovsExternalIDEntityNamespace] = entityNamespace
+	}
+	if len(ips) != 0 {
+		attachInfo[ovsExternalIDIPs] = strings.Join(ips, ipsSplitter)
```

Comment:
The ovsExternalIDIPs is set according to ExternalNode.interfaces[0].IPs. We don't require it is consistent with the IPs configured on the link, e.g. there are multiple IPs on the link, but user only configures one of them for ExternalNode.interfaces[0]. If we do not cache the IPs, it will cause the InterfaceConfig for the ExternalEntity which is loaded from OVSDB has no IPs, and we will need to wait ExternalNodeController reconciliation process to set IPs. Do you prefer to remove ovsExternalIDIPs from attchInfo?

Reply:
attachInfo here is the same as externalIDs. They are consistent.

Reply:
I meant https://github.com/antrea-io/antrea/blob/684dca3e30a333ff0653ea54df4060ae9e13406e/pkg/agent/cniserver/pod_configuration.go#L50

Reply:
Got it, done.